design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,306 @@
+  that has already graduated.
+  > **Note**: Removal of a GA feature is considered an exception.
+  > Strong arguments and a wide agreement is required for such an action.
+- As a KubeVirt cluster operator, I would like to experiment with
+  a newly-proposed ("Alpha") feature in a controlled environment, to see
+  if it makes sense to me.
+- As a KubeVirt cluster operator, I would like to evaluate an
+  undergraduate ("Beta") feature in a real-life environment with actual users.
+- As a KubeVirt cluster operator, I would like to keep using a feature
+  that got graduated after I used it during the evaluation period.
+- As a KubeVirt cluster operator, I would like to know that
+  an undergraduate feature is planned to be removed.
+- As a KubeVirt cluster operator, I would like to stop using a feature
+  that got removed.
+
+### Repos
+This is a cross repo project policy under the
+[kubevirt](https://github.com/kubevirt) organization.
+
+## Proposal Design
+The proposal on how to define a feature lifecycle is influenced by
+processes and policies from the Kubernetes project.
+These sources are scattered around, each focusing on different
+aspects of a feature:
+- [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+- [Graduation](https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/)
+- [Changing the API](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md)
+- [Deprecation](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+The proposal takes the top-down approach, starting with the high level
+flow that a common feature will traverse through.
+Continuing with actions that need to be taken and timeline suggestions.
+
+Both feature graduation and discontinuation flows are covered.
+Including the implications on users.
+
+Depending on individual topics, follow-up proposals may extend the basic
+points raised in this proposal.
+
+### Feature Stages
+A feature is expected to pass through the following stages:
+1. Enhancement proposal.
+2. Implementation.
+3. Release as Alpha (experimental).
+4. Release as Beta (pre-release for evaluation).
+5. Release as General Availability (graduation).
+6. Removal.
+
+Starting from the Alpha release, it can be removed with restrictions that
+depend on the release stage (Alpha, Beta, GA).
+
+[Removal](#removal) of features is widely discussed later
+in this proposal.
+
+#### Enhancement proposal
+As the first step for introducing a new feature, a formal proposal is
+expected to be shared for public review via mailinglist and
+a [design proposal](https://github.com/kubevirt/community/tree/main/design-proposals).
+
+This is the first opportunity to evaluate a new feature.
+The proposal needs to include motivation, goals, implementation details
+and phases. Review the [proposal template](https://github.com/kubevirt/community/blob/main/design-proposals/proposal-template.md)
```

Comment:
Kuberentes KEP template has a section for `Graduation Criteria`. I wonder if this should be included in the design template for kubevirt design docs?

My understanding of the section is that it allows for developers to state what the feature's GA will look like. It helps develoopers/users/cluster-admins align on the feature's implementation during evaluation

Reply:
All said in this thread is very important.
I think it should be discussed in it own proposal/PR, probably by suggesting changes to the proposal template.

Reply:
I have opened https://github.com/kubevirt/community/issues/286 to track this.

Resolving this thread.